helix-core/src/increment/number.rs

```diff
@@ -56,9 +55,53 @@ impl<'a> NumberIncrementor<'a> {
             (10, false)
         };
 
-        let number = if prefixed { &word[2..] } else { &word };
+        let number = if prefixed {
+            word[2..].replace('_', "")
+        } else {
+            if !word_range.contains(range.head) {
```

Comment:
This check is false, when using single digit numbers.

Reply:
having a closer look.. with a debugger.. it appears the running application does not send `Range::point` ranges.. but real ranges.. a cursor on `point(0)` results in a `range(0,1)`.